modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/near/consistency/IgniteConsistencyViolationException.java

```diff
@@ -17,20 +17,35 @@
 
 package org.apache.ignite.internal.processors.cache.distributed.near.consistency;
 
+import java.util.Set;
 import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.internal.processors.cache.KeyCacheObject;
 
 /**
- * Possible consistency violation exception.
- * Each such case should be rechecked under locks.
+ * Consistency violation exception.
  */
 public class IgniteConsistencyViolationException extends IgniteCheckedException {
     /** */
     private static final long serialVersionUID = 0L;
 
+    /** Inconsistent entries keys. */
+    private final Set<KeyCacheObject> keys;
+
+    /**
+     * @param keys Keys.
+     */
+    public IgniteConsistencyViolationException(Set<KeyCacheObject> keys) {
+        super("Distributed cache consistency violation detected.");
+
+        assert keys != null && !keys.isEmpty();
+
+        this.keys = keys;
+    }
+
     /**
-     * @param msg Message.
+     * Found but unrepaired (because of chosen strategy) inconsistent entries keys.
```

Comment:
A little bit misleading. Keys aren't repaired because Ignite hasn't tried to repair them. Ignite throws `IrreparableException` if it didn't repair keys because of strategy, don't it?

Reply:
Possible cases are 
1) CHECK_ONLY - always add all broken keys to this set.
2) LWW and RELATIVE_MAJORITY, where some or all keys can not be repaired.